src/main/java/com/mineboundteam/minebound/entity/MyrialSwordEntity.java

```diff
@@ -0,0 +1,92 @@
+package com.mineboundteam.minebound.entity;
+
+import com.mineboundteam.minebound.entity.registry.EntityRegistry;
+import com.mineboundteam.minebound.item.registry.ItemRegistry;
+import net.minecraft.world.damagesource.DamageSource;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.EntityType;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.entity.projectile.ThrowableItemProjectile;
+import net.minecraft.world.item.Item;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.Blocks;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.phys.Vec3;
+import org.jetbrains.annotations.NotNull;
+
+public class MyrialSwordEntity extends ThrowableItemProjectile {
+    ItemStack myrialSwordVacuum;
+
+    public MyrialSwordEntity(EntityType<MyrialSwordEntity> entityType, Level level) {
+        super(entityType, level);
+    }
+
+    public MyrialSwordEntity(Player player, Level level, ItemStack myrialSwordVacuum) {
+        super(EntityRegistry.MYRIAL_SWORD_ENTITY.get(), player, level);
+        this.myrialSwordVacuum = myrialSwordVacuum;
+    }
+
+    @Override
+    protected @NotNull Item getDefaultItem() {
+        return ItemRegistry.MYRIAL_SWORD.get();
+    }
+
+    @Override
+    public boolean isInWater() {
+        return false;
+    }
+
+    @Override
+    public boolean isNoGravity() {
+        return true;
+    }
+
+    @Override
+    public void onInsideBlock(@NotNull BlockState blockState) {
+        if (myrialSwordVacuum != null && !blockState.is(Blocks.AIR) && !myrialSwordVacuum.getOrCreateTag().getBoolean("minebound.return_myrial_sword"))
+            setDeltaMovement(Vec3.ZERO);
+    }
+
+    @Override
+    public void tick() {
+        move();
+        checkInsideEntity();
+        checkInsideBlocks();
+        super.tick();
+    }
+
+    // helper methods
+    public void move() {
```

Comment:
After doing Telekinetic defensive, I've learned Minecraft already has a bunch of Move methods for entities. You should be able to use the existing `move` method alongside the velocity. See the Telekinetic Defensive code for an example